aten/src/THCUNN/MultiLabelMarginCriterion.cu

```diff
@@ -131,7 +131,6 @@ __global__ void cunn_MultiLabelMarginCriterion_updateGradInput_kernel(Dtype *gra
         }
       }
     }
-    __syncthreads();
```

Comment:
I am very sorry, but I think this is actually needed. Since this `reduceBlock` is in a loop, this prevents non-0 threads override shared memory in the next loop before thread 0 reads from it. Since there are no other `__syncthreads` in this loop, we can only rely on this one. So in fact, the comments I added in my PR about "no need to call __syncthreads` before reduceBlock" etc. are misleading. We should fix those as well.

Reply:
Yikes, good point. Nice catch!